tree/ntuple/v7/inc/ROOT/RPageStorage.hxx

```diff
@@ -194,6 +194,20 @@ public:
    using Callback_t = std::function<void(RPageSink &)>;
 
 protected:
+   /// Input for the SealPage() method
+   struct RSealPageConfig {
+      const RPage &fPage; ///< input page to be sealed
+      const RColumnElementBase &fElement; ///< Corresponds to the page's elements, used for size calculation etc.
+      int fCompressionSetting = 0; ///< Compression algorithm and level to apply
+      /// If false, the output buffer must not point to the input page buffer, which would otherwise be an option
+      /// if the page is mappable and should not be compressed
+      bool fAllowAlias = true;
+      /// Location for sealed output. The memory buffer has to be large enough.
+      void *fBuffer = nullptr;
```

Comment:
If the buffer is mandatory, maybe it'd be better to add it as a ctor param?

Reply:
So that would be https://github.com/root-project/root/pull/15894/commits/9a99a4eef64f2f3fce75e12e45fbdf5898c364dd

Reply:
In that case I would suggest asserting that all the expected parameters are there at the beginning of `SealPage()`, so if someone forgets one it's easier to see immediately.